### PR TITLE
Update Connect Analytics charts

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR enhances analytics dashboard components with new chart configurations, wallet statistics, and story utilities.

### Detailed summary
- Exported `WalletStats` interface for reusability.
- Updated CSS variables for chart styling.
- Refactored chart components and story utilities.
- Added new chart cards for daily connections, wallet connectors, and wallet distribution.
- Implemented loading and empty state components for charts.

> The following files were skipped due to too many changes: `apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/ConnectAnalyticsDashboard.tsx`, `apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/DailyConnectionsChartCard.tsx`, `apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/WalletDistributionChartCard.tsx`, `apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/WalletConnectorsChartCard.tsx`, `apps/dashboard/src/pages/dashboard/connect/analytics.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by MananTank** [comment]
* **#4372** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4372?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MananTank and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/pages/dashboard/connect/analytics.tsx [hunk 1]
```diff
@@ -5,57 +5,30 @@ import {
...
-  Legend,
-  Pie,
-  PieChart,
-  ResponsiveContainer,
-  Sector,
-} from "recharts";
-import { Card, Heading, LinkButton, Text } from "tw-components";
+import { SiUnrealengine } from "react-icons/si";
+import { SiDotnet } from "react-icons/si";
+import {} from "tw-components";
```
Comment (jnsdls) [inline apps/dashboard/src/pages/dashboard/connect/analytics.tsx:20]: remove

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/DailyConnectionsChartCard.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  const chartLabelToShow: Record<ChartToShow, string> = {
+    uniqueWallets: "Unique Wallets",
+    totalWallets: "Total Wallets",
+  };
+
+  type ChartData = {
+    time: string; // human readable date
+    totalWallets: number;
+    uniqueWallets: number;
+  };
```
Comment (jnsdls) [inline apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/DailyConnectionsChartCard.tsx:50]: move out of the component itself (no need to re-instantiate on every render)

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/DailyConnectionsChartCard.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import type { WalletStats } from "@3rdweb-sdk/react/hooks/useApi";
+import { format } from "date-fns";
+import { useMemo, useState } from "react";
+import { Bar, BarChart, CartesianGrid, LabelList, XAxis } from "recharts";
+import { EmptyChartState, LoadingChartState } from "./EmptyChartState";
+
+type ChartToShow = "uniqueWallets" | "totalWallets";
```
Comment (jnsdls) [inline apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/DailyConnectionsChartCard.tsx:20]: could this just be `typeof chartToShowoptions[number]` ?

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/WalletConnectorsChartCard.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  const chartToShowOptions: ChartToShow[] = ["uniqueWallets", "totalWallets"];
+
+  const chartLabelToShow: Record<ChartToShow, string> = {
+    uniqueWallets: "Unique Wallets",
+    totalWallets: "Total Wallets",
+  };
+
+  type ChartData = Record<string, number> & {
+    time: string; // human readable date
+  };
```
Comment (jnsdls) [inline apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/WalletConnectorsChartCard.tsx:41]: move up

**Review by jnsdls — CHANGES_REQUESTED** [review]

**Review by MananTank — COMMENTED** [review]
Fix the Client ID selector position on mobile

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/DailyConnectionsChartCard.tsx (hunk 2 above)
Comment (MananTank) [reply apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/DailyConnectionsChartCard.tsx:50]: It's just a type, but sure

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/DailyConnectionsChartCard.tsx (hunk 3 above)
Comment (MananTank) [reply apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/DailyConnectionsChartCard.tsx:20]: Not sure what you mean, `ChartToShow` is the source type and everything else is defined using it

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/DailyConnectionsChartCard.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  isLoading: boolean;
+}) {
+  const { walletStats } = props;
+
+  const [chartToShow, setChartToShow] = useState<ChartToShow>("uniqueWallets");
+  const chartToShowOptions: ChartToShow[] = ["uniqueWallets", "totalWallets"];
+  const chartLabelToShow: Record<ChartToShow, string> = {
+    uniqueWallets: "Unique Wallets",
+    totalWallets: "Total Wallets",
+  };
```
Comment (jnsdls) [inline apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/DailyConnectionsChartCard.tsx:50]: these are static, too. move them out?

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/WalletConnectorsChartCard.tsx [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+  isLoading: boolean;
+}) {
+  const { walletStats } = props;
+  const [chartToShow, setChartToShow] = useState<ChartToShow>("uniqueWallets");
+  const chartToShowOptions: ChartToShow[] = ["uniqueWallets", "totalWallets"];
+
+  const chartLabelToShow: Record<ChartToShow, string> = {
+    uniqueWallets: "Unique Wallets",
+    totalWallets: "Total Wallets",
+  };
```
Comment (jnsdls) [inline apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/WalletConnectorsChartCard.tsx:41]: static, move them out of the component

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/WalletDistributionChartCard.tsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+  isLoading: boolean;
+}) {
+  const { walletStats } = props;
+  const [chartToShow, setChartToShow] = useState<ChartToShow>("uniqueWallets");
+  const chartToShowOptions: ChartToShow[] = ["uniqueWallets", "totalWallets"];
+
+  const chartLabelToShow: Record<ChartToShow, string> = {
+    uniqueWallets: "Unique Wallets",
+    totalWallets: "Total Wallets",
+  };
```
Comment (jnsdls) [inline apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/WalletDistributionChartCard.tsx:42]: static, move up

**Review by jnsdls — CHANGES_REQUESTED** [review]

**Review by jnsdls — APPROVED** [review]

**Pull request merged**